api/v1alpha1/tls_types.go

```diff
@@ -133,3 +147,36 @@ type ClientValidationContext struct {
 	// +optional
 	CACertificateRefs []gwapiv1.SecretObjectReference `json:"caCertificateRefs,omitempty"`
 }
+
+// StatelessSessionResumptionSettings defines setting for stateless (session-ticket based) session resumption
+// +kubebuilder:validation:XValidation:rule="(self.enabled == false && self.sessionTicketKeysRef.size() > 0)",message="sessionTicketKeysRef can only be set when enabled is true"
+type StatelessSessionResumptionSettings struct {
+	// Enabled determines if stateless session resumption is supported. When Envoy Proxy is deployed with more than
+	// one replica, session ticket encryption keys are not synchronized between instances, possibly leading to resumption
+	// failures. Users can synchronize session ticket encryption keys be storing them in a secret and using the
+	// sessionTicketKeysRef option.
+	// Note that improper handling of session ticket encryption keys may result in loss of secrecy.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-tlssessionticketkeys
+	// Default: true
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// SessionTicketKeysRef is a reference to the secret containing the keys to use for encryption and decryption
+	// of session tickets. The first keys is used for encryption and all keys are candidate for decryption.
+	// Users are responsible for securely generating, storing and rotating keys.
+	// If empty, each Envoy Proxy instance will use local in-memory keys.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-tlssessionticketkeys
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	SessionTicketKeysRef []gwapiv1.SecretObjectReference `json:"sessionTicketKeysRef,omitempty"`
```

Comment:
Should we mention the risks of the shared ticket secret and suggestions to mitigate the risks? Like Envoy does: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-tlssessionticketkeys

Reply:
after discussion yesterday, I reduced the scope of this PR to only deal with enable/disable of resumption.